usercp.php

```diff
@@ -3221,15 +3221,18 @@
 
 			$db->insert_query("joinrequests", $joinrequest);
 
-			foreach($groupleaders[$usergroup['gid']] as $leader)
+			if(in_array($usergroup['gid'], $groupleaders))
```

Comment:
A simple script on php 5.6:

```php
<?php

$array = [1 => [2 => ['uid' => 2, 'username' => 'test']]];

echo "<pre>";
var_dump(in_array(1, $array)); // False
var_dump(in_array(2, $array)); // False
var_dump(is_array($array[1])); // True
var_dump(is_array($array[2])); // False (and a notice)
echo "</pre>";
```

As you can see `in_array` shows the wrong result.